### DISCUSSION TITLE
Specific vitest environment for specific tests

### DISCUSSION THREAD

**Discussion by lcoq** [vitest-dev/vitest#4968] [Q&A] [answered]
In a React app, I would like to use `happy-dom` on some tests, but switch to `jsdom` when it requires features not available in happy-dom.

I saw [there is a `@vitest-environment` config for specific files](https://vitest.dev/guide/environment#environments-for-specific-files), but is there any way to achieve this on specific tests instead ?

Thanks !

**Comment by sheremet-va** [comment, marked as answer]
The linked page has this sentence:

> Or you can also set [environmentMatchGlobs](https://vitest.dev/config/#environmentmatchglobs) option specifying the environment based on the glob patterns.

Is this what you are looking for? 

You cannot change specific test functions because the whole file runs in a specified environment.

**Reply by lcoq** [reply]
Ok I understand, I'll use different files then ! Thank you for your answer

**Comment by Hugos68** [comment]
I'm looking to use different environments in different tests in the same file, the comment or globs don't help since they only apply on a file basis.
